client/src/app/s-degree-plan/s-degree-plan.component.ts

```diff
@@ -14,16 +14,25 @@ import { share } from "rxjs/operators";
 })
 export class SDegreePlanComponent implements OnInit {
   yearArray: Observable<Array<Year>>;
+  openPanel = false;
 
   constructor(
     private router: Router,
     private errorHandler: ErrorHandler,
     private planService: PlanService
   ) {
     this.yearArray = this.planService.formatPlan();
```

Comment:
Correct me if I'm wrong, but shouldn't the constructor be used only for dependency injection and initializing default values for the component's variables? I've read in angular that it's good practice to put complex initializations on ngOnInit instead so that the component is "easier to build"- whatever that means.

https://angular.io/guide/lifecycle-hooks#oninit

Reply:
You are probably correct, I can move the initialization to onInit()

Reply:
Hm... what I meant was that the constructor should only have simple initialization of the component's variables to default values. 

In this case, the constructor should set the openPanel and yearArray variables to "true" and an empty array respectively. 

The ngOnInit then handles the more complex initialization, which is calling the formatPlan() function of planService.